openml/datasets/dataset.py

```diff
@@ -773,6 +802,40 @@ def get_data(
 
         return data, targets, categorical, attribute_names
 
+    def _load_metadata(self, features: bool = False, qualities: bool = False):
+        """Load the missing metadata information from the server and store it in the
+        dataset object.
+
+        The purpose of the function is to support lazy loading.
+
+        Parameters
+        ----------
+        features : bool (default=False)
+            If True, load the `self.features` data if not already loaded.
+        qualities: bool (default=False)
+            If True, load the `self.qualities` data if not already loaded.
+        """
+        # Delayed Import to avoid circular imports or having to import all of dataset.functions to
+        # import OpenMLDataset
+        from openml.datasets.functions import _get_dataset_metadata
+
+        if self.dataset_id is None:
+            raise ValueError(
+                """No dataset id specified. Please set the dataset id.
+                                Otherwise we cannot load metadata."""
+            )
+
+        features_file, qualities_file = _get_dataset_metadata(
+            self.dataset_id, features=features, qualities=qualities
+        )
+
+        if features_file is not None:
+            self._features = _read_features(features_file)
+
+        if qualities_file is not None:
+            self._qualities = _read_qualities(qualities_file)
+            self._no_qualities_found = self._qualities is None
```

Comment:
This is required as far as I can tell, because `self._no_qualities_found` is initially False and only becomes True in this edge case after lazy loading the qualities for the first time.